kettle-plugins/hdfs/src/main/java/org/pentaho/big/data/kettle/plugins/hdfs/trans/HadoopFileInputMeta.java

```diff
@@ -219,14 +234,50 @@ public FileInputList getFileInputList( VariableSpace space ) {
         inputFiles.fileName[ i ] = "";
       }
     }
-    return createFileList( space );
+    FileInputList returnList = createFileList( bowl, space );
+    for ( int i = 0; i < inputFiles.fileName.length; i++ ) {
+      if ( !canAccessHdfs( inputFiles.fileName[ i ], fatalErrorOnHdfsNotFound ) ) {
+        returnList.addNonAccessibleFile( new NonAccessibleFileObject( inputFiles.fileName[ i ] ) );
+      }
+    }
+    return returnList;
   }
 
   /**
-   * Created for test purposes
+   * If the KETTLE_FATAL_ERROR_ON_HDFS_NOT_FOUND property is set to Y, return false if we can find a named cluster that should
+   * be used to access the file AND there is no corresponding HDFS file system for that named cluster.
+   *
+   * @param fileName
+   * @return false if the filename should be accessed via a named cluster and HDFS and it cannot and the KETTLE_FATAL_ERROR_ON_HDFS_NOT_FOUND
+   * property is Y
    */
+  protected boolean canAccessHdfs( String fileName, boolean checkHdfs ) {
+    if ( checkHdfs ) {
+      try {
+        URI fileUri = new URI( fileName );
+        NamedCluster c = namedClusterService.getNamedClusterByHost( fileUri.getHost(), getParentStepMeta().getParentTransMeta().getMetaStore() );
+        if ( null == c && NAMED_CLUSTER_SCHEME.equalsIgnoreCase( fileUri.getScheme() ) ) {
+          c = namedClusterService.getNamedClusterByName( fileUri.getHost(), getParentStepMeta().getParentTransMeta().getMetaStore() );
+        }
+        if ( null != c && null == hadoopFileSystemLocator.getHadoopFilesystem( c, fileUri ) ) {
+          return false;
+        }
+      } catch ( URISyntaxException | ClusterInitializationException e ) {
+        return false;
+      }
+    }
+    return true;
+  }
+
   FileInputList createFileList( VariableSpace space ) {
-    return FileInputList.createFileList( space, inputFiles.fileName, inputFiles.fileMask, inputFiles.excludeFileMask,
+    return createFileList( null, space );
+  }
+
+  /**
+   * Created for test purposes
+   */
+  FileInputList createFileList( Bowl bowl, VariableSpace space ) {
+    return FileInputList.createFileList( bowl, space, inputFiles.fileName, inputFiles.fileMask, inputFiles.excludeFileMask,
       inputFiles.fileRequired, inputFiles.includeSubFolderBoolean() );
```

Comment:
This will end up passing a null Bowl, which isn't correct (will probably cause an NPE). 

Either pass `DefaultBowl.getInstance()`, `getTransMeta().getBowl()` (this is ideal), or just use the method signature that doesn't include the Bowl. 

Rogue 2 will revisit all these steps for 10.3, so don't worry about passing the Bowl along, for now.